pkg/services/datasources/fakes/fake_datasource_service.go

```diff
@@ -45,6 +45,16 @@ func (s *FakeDataSourceService) GetAllDataSources(ctx context.Context, query *da
 	return s.DataSources, nil
 }
 
+func (s *FakeDataSourceService) GetPrunableProvisionedDataSources(ctx context.Context) (res []*datasources.DataSource, err error) {
+	var dataSources []*datasources.DataSource
+	for _, datasource := range s.DataSources {
+		if datasource.IsPrunable {
+			dataSources = append(dataSources, datasource)
+		}
+	}
```

Comment:
nit: consistent with the two word data source in `s.DataSources`

```suggestion
	for _, dataSource := range s.DataSources {
		if dataSource.IsPrunable {
			dataSources = append(dataSources, dataSource)
		}
	}
```